lib/rubygems/indexer.rb

```diff
@@ -135,7 +145,6 @@ def build_marshal_gemspecs
 
     Gem.time 'Generated Marshal quick index gemspecs' do
       Gem::Specification.each do |spec|
-        next if spec.default_gem?
```

Comment:
@zzak I don't think so. I've tested this approach.

This was the previous patch behavior (#661) - in this line it naively skipped every default_gem regardless of its inclusion in the directory containing the gems you want to index (the path pointed by the param `--directory`).

This patch, simply initialize the indexer's working set (`Gem::Specification#@_all`) with an empty set. This way, it ignores every default gem unless it is included in the directory containing the gems you want to index.

Do you have any example in mind?